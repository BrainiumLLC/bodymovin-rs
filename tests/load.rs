use bodymovin::Bodymovin;
use std::path::Path;

fn test_file(file: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let path = Path::new(env!("CARGO_MANIFEST_DIR"))
        .join("tests/data")
        .join(file);
    let loaded = Bodymovin::load(path).unwrap();
    insta::assert_debug_snapshot!(loaded);
}

#[test]
fn test_shapes() {
    test_file("shapes.json")
}

#[test]
fn test_4thofjuly2018() {
    test_file("4thofjuly2018.json")
}

#[test]
fn test_lottie_adrock() {
    test_file("lottie_adrock.json")
}

#[test]
fn test_lottie_bodymovin() {
    test_file("lottie_bodymovin.json")
}

#[test]
fn test_lottie_gatin() {
    test_file("lottie_gatin.json")
}

#[test]
fn test_lottie_happy2016() {
    test_file("lottie_happy2016.json")
}

#[test]
fn test_underline() {
    test_file("underline.json")
}
