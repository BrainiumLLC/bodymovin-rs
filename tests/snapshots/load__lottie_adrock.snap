---
source: tests/load.rs
expression: loaded
---
Bodymovin {
    in_point: 0.0,
    out_point: 392.0,
    frame_rate: 30.0,
    width: 690,
    is_3d: false,
    height: 913,
    version: "4.0.0",
    name: None,
    layers: [
        Shape(
            Shape {
                transform: Transform {
                    anchor_point: Property {
                        value: Fixed(
                            [
                                0.0,
                                0.0,
                                0.0,
                            ],
                        ),
                        expression: None,
                        index: None,
                    },
                    position: Property {
                        value: Fixed(
                            [
                                347.285,
                                1375.72,
                                0.0,
                            ],
                        ),
                        expression: None,
                        index: None,
                    },
                    scale: Property {
                        value: Fixed(
                            [
                                100.0,
                                100.0,
                                100.0,
                            ],
                        ),
                        expression: None,
                        index: None,
                    },
                    rotation: Property {
                        value: Fixed(
                            0.0,
                        ),
                        expression: None,
                        index: None,
                    },
                    opacity: Property {
                        value: Fixed(
                            100.0,
                        ),
                        expression: None,
                        index: None,
                    },
                    position_x: Property {
                        value: Fixed(
                            0.0,
                        ),
                        expression: None,
                        index: None,
                    },
                    position_y: Property {
                        value: Fixed(
                            0.0,
                        ),
                        expression: None,
                        index: None,
                    },
                    position_z: Property {
                        value: Fixed(
                            0.0,
                        ),
                        expression: None,
                        index: None,
                    },
                    skew: Property {
                        value: Fixed(
                            0.0,
                        ),
                        expression: None,
                        index: None,
                    },
                    skew_axis: Property {
                        value: Fixed(
                            0.0,
                        ),
                        expression: None,
                        index: None,
                    },
                },
                auto_orient: false,
                blend_mode: Normal,
                is_3d: false,
                index: 0,
                html_class: None,
                html_id: None,
                in_point: 339.0,
                out_point: 392.0,
                start_time: 0.0,
                name: Some(
                    "Shape Layer 2",
                ),
                masks: [],
                effects: [],
                stretch: 1.0,
                parent: Some(
                    2,
                ),
                shapes: [],
            },
        ),
    ],
}
